p2p/src/config.rs

```diff
@@ -29,7 +29,7 @@ make_config_setting!(NodeTypeSetting, NodeType, NodeType::Full);
 make_config_setting!(AllowDiscoverPrivateIps, bool, false);
 make_config_setting!(PingCheckPeriod, Duration, Duration::from_secs(60));
 make_config_setting!(PingTimeout, Duration, Duration::from_secs(150));
-make_config_setting!(MaxClockDiff, Duration, Duration::from_secs(10));
+make_config_setting!(MaxClockDiff, Duration, Duration::from_secs(12));
```

Comment:
I bumped the value here a little since the new approach is slightly less forgiving. Now both endpoints of local time interval in which the request is being processed must adhere to the max limit. It only checked one end of the interval before.

Reply:
Do we want to make this smaller now?

Reply:
Hmm I thought I reverted it. Must be a merge issue

Reply:
Reverted to 10s and removed some stuff that was mistakenly introduced by a merge. Do we want to go below 10s?

Reply:
That's an excellent question. I would go to 5 seconds, but I'd rather we do some research before making it this tight, and try to understand whether connections around the world can take this long under normal circumstances.

Reply:
> That's an excellent question. I would go to 5 seconds, but I'd rather we do some research before making it this tight, and try to understand whether connections around the world can take this long under normal circumstances.

The tolerance for connection latency is included in the handshake timeout so this value is the tolerance for clock differences _on top of_ latency. Tuning how strict our requirements are with respect to connection quality should be done by adjusting the handshake timeout parameter rather than this one. The effective tolerance is `handshake_timeout + max_clock_diff`.

Now this could be changed so `max_clock_diff` specifies the diff _including_ network latency, i.e. the effective tolerance mentioned in the previous paragraph. The interval of allowable peer time would be adjusted by `new_max_clock_diff - handshake_timeout`. Basically just shift the value by handshake timeout. There are tradeoffs:
* Pro: the semantics of `max_clock_diff` becomes clearer: it specifies by how much can peer's clock differ from ours.
* Con: Values smaller than handshake timeout no longer make sense. Would have to be handled somehow.
* Pro: When adjusting handshake timeout without changing effective latency does not involve changing the other parameter
* Con: When adjusting handshake timeout in a way where we want the max clock diff on top of latency remain constant requires also adjusting this parameter accordingly.

The last two bullets are the other way around in the current implementation. Seems there is no ideal solution here.

Reply:
We can put both variables in a struct and manage them together in a constructor to limit the cases where they don't make sense.

Reply:
> We can put both variables in a struct and manage them together in a constructor to limit the cases where they don't make sense.

That would be a nice implementation approach but there is still the question of which one to present to the user as the config option.